pkg/controller/core/clusterqueue_controller.go

```diff
@@ -497,3 +552,62 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 	}
 	return nil
 }
+
+func (r *ClusterQueueReconciler) getWorkloadsStatus(cq *kueue.ClusterQueue) *kueue.ClusterQueuePendingWorkloadsStatus {
+	if r.shouldUpdatePendingWorkloadStatus(cq) {
+		return &kueue.ClusterQueuePendingWorkloadsStatus{
+			Head:           r.qManager.GetSnapshot(cq.Name),
+			LastChangeTime: metav1.Time{Time: time.Now()},
+		}
+	}
+	return cq.Status.PendingWorkloadsStatus
+}
+
+func (r *ClusterQueueReconciler) shouldUpdatePendingWorkloadStatus(cq *kueue.ClusterQueue) bool {
+	return cq.Status.PendingWorkloadsStatus == nil ||
+		time.Since(cq.Status.PendingWorkloadsStatus.LastChangeTime.Time) >= r.queueVisibilityUpdateInterval &&
+			!equality.Semantic.DeepEqual(cq.Status.PendingWorkloadsStatus.Head, r.qManager.GetSnapshot(cq.Name))
+}
+
+func (r *ClusterQueueReconciler) Start(ctx context.Context) error {
+	defer r.snapshotsQueue.ShutDown()
```

Comment:
Skip the body of the function when the `MaxCount = 0` (the feature is disabled)`. It would be also good to have a dedicated integration test for that case.

Reply:
Actually, we disable the feature earlier: https://github.com/kubernetes-sigs/kueue/pull/1069/files#diff-a59d3d6d0bd5d684e293231de0698d9af2df82e7373cfb898dd8b0d89f8ffe87R53
But, if you would like to move that condition here it's okey to me. It will just add an extra registration to the manager which is what I wanted to avoid

Reply:
I see. Still, if not too much effort, I would prefer to move it. Because the Start can be doing potentially other things as well.

Reply:
moved here https://github.com/kubernetes-sigs/kueue/pull/1069/commits/90b18bdee0de5861442bc95320133618fbf3ced0